libevmasm/Assembly.cpp

```diff
@@ -273,6 +273,9 @@ Json::Value Assembly::assemblyJSON(std::map<std::string, unsigned> const& _sourc
 		Json::Value& jsonSourceList = root["sourceList"];
 		for (auto const& [name, index]: _sourceIndices)
 			jsonSourceList[index] = name;
+		for (unsigned i = 0 ; i < _sourceIndices.size(); ++i)
+			if (jsonSourceList[i] == Json::nullValue)
+				jsonSourceList[i] = "unknown-source-" + std::to_string(i);
```

Comment:
I don't think so. The problem is that the `nullValue` will only be added to the `jsonSourceList` when `index` (returned by `_sourceIndices`) is greater than the expected number of items. For instance when you compile multiple sources but only export one. But I just realized that the logic is indeed wrong anyway, because the second loop should go until the maximum `index` value and not the size of `_sourceIndices`. The whole idea is to fill the gaps on the indices that we may have when exporting assembly, since it may not contain all the sources information.